src/main/java/com/amazon/opendistro/elasticsearch/performanceanalyzer/decisionmaker/actions/TimedFlipFlopDetector.java

```diff
@@ -0,0 +1,114 @@
+package com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions;
+
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.collections.TimeExpiringSet;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Dimension;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.decisionmaker.actions.ImpactVector.Impact;
+import com.amazon.opendistro.elasticsearch.performanceanalyzer.rca.store.rca.cluster.NodeKey;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * A {@link FlipFlopDetector} whose recorded actions expire after a given period of time.
+ *
+ * <p>This class defines a flip flop as an {@link Impact#DECREASES_PRESSURE}s followed by an
+ * {@link Impact#INCREASES_PRESSURE}s to be a flip flops.
+ *
+ * <p>This class stores a {@link TimeExpiringSet} of {@link ImpactVector}s per {@link NodeKey}
+ * that are used to determine these flip flops.
+ */
+public class TimedFlipFlopDetector implements FlipFlopDetector {
+    private Map<NodeKey, TimeExpiringSet<ImpactVector>> flipFlopMap;
+    private long expiryDuration;
+    private TimeUnit expiryUnit;
+
+    public TimedFlipFlopDetector(long duration, TimeUnit unit) {
+        flipFlopMap = new HashMap<>();
+        this.expiryDuration = duration;
+        this.expiryUnit = unit;
+    }
+
+    /**
+     * Tests if (a,b) is a flip flopping sequence of impacts.
+     *
+     * <p>Only an increase following a decrease is considered a flip flop
+     *
+     * @param a The first impact that would be applied
+     * @param b The subsequent impact that would be applied
+     * @return Whether or not (a,b) is a flip flopping sequence of impacts
+     */
+    protected boolean isFlipFlopImpact(Impact a, Impact b) {
```

Comment:
Rename `a -> prev` and `b -> curr` or something on those lines to make it less ambiguous?